src/main/java/org/fungover/haze/Main.java

```diff
@@ -4,29 +4,59 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
 
 public class Main {
-    public static void main(String[] args) {
-
-        try (ServerSocket serverSocket = new ServerSocket(6379)) {
-            while (true) {
-                var client = serverSocket.accept();
-
-                BufferedReader input = new BufferedReader(new InputStreamReader(client.getInputStream()));
-                System.out.println(input.readLine());
-                System.out.println(input.readLine());
-                System.out.println(input.readLine());
-                System.out.println(input.readLine());
-                System.out.println(input.readLine());
-                System.out.println(input.readLine());
-                System.out.println(input.readLine());
-
-                client.getOutputStream().write("+OK\r\n".getBytes());
-
-                client.close();
-            }
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-    }
+	public static void main(String[] args) {
+
+		try (ServerSocket serverSocket = new ServerSocket(6379)) {
+			while (true) {
+				var client = serverSocket.accept();
+
+				Runnable newThread = () -> {
+					try {
+
+
+						BufferedReader input = new BufferedReader(new InputStreamReader(client.getInputStream()));
+						System.out.println(input.readLine());
+						System.out.println(input.readLine());
+						System.out.println(input.readLine());
+						System.out.println(input.readLine());
+						System.out.println(input.readLine());
+
+
+						printThreadDebug();
+
+						Lock lock = new ReentrantLock();
+						lock.lock();
+						try{
+							// Edit Database here
+						}finally {
+							lock.unlock();
+						}
```

Comment:
This code for looking access to database can be moved to #55 and use the same lock object for all methods in the database.

Reply:
Lock removed from this issue